plugins/guests/linux/cap/public_key.rb

```diff
@@ -53,6 +53,7 @@ def self.remove_public_key(machine, contents)
               result=$?
             fi
             rm -f '#{remote_path}'
+            chmod 0600 ~/.ssh/authorized_keys
```

Comment:
If possible, could you please provide the debug output for when it fails? Thanks!